subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1387,29 +1432,51 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder, uint16_t laz
 	if (!lll->is_hdcd)
 #endif /* CONFIG_BT_PERIPHERAL */
 	{
-		uint32_t random_delay;
 		uint32_t ret;
 
-		lll_rand_isr_get(&random_delay, sizeof(random_delay));
-		random_delay %= ULL_ADV_RANDOM_DELAY;
-		random_delay += 1;
+		curr_random_delay = adv->next_random_delay;
+
+		lll_rand_isr_get(&adv->next_random_delay, sizeof(adv->next_random_delay));
+		adv->next_random_delay %= ULL_ADV_RANDOM_DELAY;
+		adv->next_random_delay += 1;
 
 		ret = ticker_update(TICKER_INSTANCE_ID_CTLR,
 				    TICKER_USER_ID_ULL_HIGH,
 				    (TICKER_ID_ADV_BASE +
 				     ull_adv_handle_get(adv)),
-				    random_delay,
+				    curr_random_delay,
 				    0, 0, 0, 0, 0,
 				    ticker_op_update_cb, adv);
 		LL_ASSERT((ret == TICKER_STATUS_SUCCESS) ||
 			  (ret == TICKER_STATUS_BUSY));
 	}
 
-#if defined(CONFIG_BT_CTLR_ADV_EXT) && (CONFIG_BT_CTLR_ADV_AUX_SET > 0)
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	ull = &adv->ull;
+
+	adv->event_counter += (lazy + 1);
+
+	interval = HAL_TICKER_US_TO_TICKS
+		((uint64_t)adv->interval * 625U);
+
+	ticks_next_evt_done = 2*interval
+		+ curr_random_delay + adv->next_random_delay;
+
+	if (adv->remain_duration) {
+		if (adv->remain_duration >= ticks_next_evt_done) {
+			adv->remain_duration -= ticks_next_evt_done;
+		} else {
+			/* A value of 1 means send term event */
+			adv->remain_duration = 1;
+		}
+	}
```

Comment:
Simply these based on use of 625us units. Using a value of 1 is unnecessary, a check `remain <= interval` in the event done would suffice, if `remain` has already been pre-deducted of the random_delay.

Reply:
The value 1 means that rx_demux_event_done will disable set, then set this to 0. If rx_demux_event_done used 0 as a criteria, it would happen when timeout is not enabled from host (duration=0).